xdsl/parser/core.py

```diff
@@ -660,9 +660,23 @@ def parse_optional_attr_dict_with_keyword(
         dictionary, and usually correspond to the names of the attributes that are
         already passed through the operation custom assembly format.
         """
-        begin_pos = self.lexer.pos
         if self.parse_optional_keyword("attributes") is None:
             return None
+        return self.parse_optional_attr_dict_with_reserved_attr_names(
+            reserved_attr_names
+        )
+
+    def parse_optional_attr_dict_with_reserved_attr_names(
+        self, reserved_attr_names: Iterable[str] = ()
+    ) -> DictionaryAttr | None:
+        """
+        Parse a dictionary attribute if present.
+        This is intended to be used in operation custom assembly format.
+        `reserved_attr_names` contains names that should not be present in the attribute
+        dictionary, and usually correspond to the names of the attributes that are
+        already passed through the operation custom assembly format.
+        """
+        begin_pos = self.lexer.pos
```

Comment:
slight change in behaviour, but I think it might be a bit more helpful, since it'll point to the dictionary directly